### PR TITLE
feat(fix): honor insecure-skip-tls-verify flag for k3s clusters.

### LINKED ISSUE #16210: Continue to inititiate connection with a warning  incase of tls certificate issues.
### Current Behavior

When connecting Meshery to a k3s cluster, TLS verification fails due to k3s generating certificates valid only for `localhost` and private IPs. This leads to a certificate mismatch when Meshery attempts to connect via the public IP or hostname.

#### **Error**

```
x509: certificate is valid for 127.0.0.1, not <ec2-public-ip>
```

#### **Root Cause**

By default, `k3s` generates TLS certificates with SANs limited to:

* `127.0.0.1`
* private/internal IPs

When remote access is needed, `k3s` must be installed with the `--tls-san <public-ip>` flag to include the external IP or hostname in the certificate.

#### **Additional Context**

* The cluster installation likely uses **self-signed certificates**.
* Unless the CA is trusted or the `kubeconfig` is adjusted to use the internal IP/hostname, TLS validation fails.
* Meshery should continue communication with such clusters by respecting the `insecure-skip-tls-verify` flag defined in the `kubeconfig` context.

#### **Expected Behavior**

Meshery should:

* Honor the `insecure-skip-tls-verify` flag in the kubeconfig.
* Proceed with connecting even when the cluster uses a self-signed or mismatched certificate.

### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://meshery.io/community#handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://meshery.io/community#community-forums) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by JunaidAli-dev**
**Notes for Reviewers**

- This PR fixes #16210 
### Current Behavior
When connecting Meshery to k3s clusters with self-signed certificates, TLS verification fails with error: x509: certificate is valid for 127.0.0.1, not <public-ip> even when `insecure-skip-tls-verify: true` is set in kubeconfig.

This occurs because the Kubernetes client-go library throws a conflict error when both `certificate-authority-data` (or `CAFile`) and `insecure-skip-tls-verify: true` are present in the kubeconfig.

### Desired Behavior
Meshery should properly honor the `insecure-skip-tls-verify` flag and successfully connect to k3s clusters without TLS verification errors.

### Implementation
Modified `getK8SClientSet()` function in `server/helpers/helpers.go` to explicitly clear CA certificate data when `insecure-skip-tls-verify` is detected in the kubeconfig.

**Code Changes:**

// After building client config, check and handle insecure flag
if clientConfig.TLSClientConfig.Insecure {
clientConfig.TLSClientConfig.CAFile = ""
clientConfig.TLSClientConfig.CAData = nil
}

This ensures that when the insecure flag is set to `true`, the CA data is cleared to prevent conflicts that cause TLS verification failures.

### Testing
Created comprehensive unit tests in `server/helpers/helpers_test.go`:

1. **TestGetK8SClientSetWithInsecureSkipVerify** - Verifies basic insecure-skip-tls-verify functionality
2. **TestGetK8SClientSetWithCADataAndInsecureSkipVerify** - Tests the conflict scenario when both CA data and insecure flag are present
3. **TestGetK8SClientSetWithContextName** - Validates context switching with insecure connections
4. **TestGetK8SClientSetWithEmptyKubeconfig** - Tests edge case handling

### Test Results
Before-
image
After-
image

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

Modify it to look more beautiful and proffesional

### CODE REVIEW

**Comment by JunaidAli-dev** [comment]
@aabidsofi19 I have updated the PR kindly have look and please review it.

File: server/helpers/kubernetes.go [hunk 1]
```diff
@@ -33,6 +33,10 @@ func getK8SClientSet(kubeconfig []byte, contextName string) (*kubernetes.Clients
 		if err != nil {
 			return nil, ErrClientConfig(err)
 		}
+		if clientConfig.TLSClientConfig.Insecure {
+			clientConfig.TLSClientConfig.CAFile = ""
+			clientConfig.TLSClientConfig.CAData = nil
+		}
```
[the remark this answers is not in the document]
Comment (lekaf974) [reply server/helpers/kubernetes.go:39]: think this comment is valid to avoid question on how we did that otherwise look good to me

**Comment by ritzorama** [comment]
It's not clear that this solves the issue, @JunaidAli-dev. Have you demonstrated this as occurring before the change, then being fixed after the change? Please provide these examples.

**Comment by JunaidAli-dev** [comment]
**@ritzorama yes it’s working now, wasn’t before the fix tho** 

image
image
